src/ros_filter.cpp

```diff
@@ -3088,6 +3096,38 @@ namespace RobotLocalization
     return retVal;
   }
 
+  template<typename T>
+  bool RosFilter<T>::validateFilterOutput(const nav_msgs::Odometry &message)
+  {
+    if (std::isnan(message.pose.pose.position.x) || std::isinf(message.pose.pose.position.x))
```

Comment:
Three things:

1. It may be a bit easier to read like this:

```
return !std::isnan(message.pose.pose.position.x) && !std::isinf(message.pose.pose.position.x) &&
  !std::isnan(message.pose.pose.position.y) && !std::isinf(message.pose.pose.position.y) &&
...
```

2. I really need to enable C++11 for Kinetic and on, because [std::isfinite](http://en.cppreference.com/w/cpp/numeric/math/isfinite) does this a bit more cleanly.

3. It'll likely only warn us a cycle or two earlier, but I wonder if we should do

```
bool is_valid = !std::isnan(message.pose.pose.position.x) && !std::isinf(message.pose.pose.position.x) &&
...
```

as above, but then check the covariances as well:

```
for (size_t i = 0; i < message.pose.covariance.size(); ++i)
{
  const double &covar = message.pose.covariance[i];
  is_valid = is_valid && !std::isnan(covar) && !std::isinf(covar);
}
```

Then we'd repeat for twist. What do you think?

Reply:
On point 1, done. 
On point 2, would you like that to be part of this PR?
On point 3, for warning a couple cycles beforehand, I think while the state is still in the set of real numbers, we should let things play out. It's a couple more checks and if its going to NaN out, the behavior I would find most helpful if I could see the process that lead there in my logs.

Reply:
No, (2) wasn't meant to be actionable. There should probably be a C++11 or C++14 PR at some point, but that's for me to worry about.

So for (3), are you advocating for checking the covariance or not? I'm really not that concerned, since we're probably look at (at most) one cycle between a `NaN/Inf` being in any part of the state or covariance and it ruining the entire thing. I'm fine to leave it as-is if you are.

Reply:
I'm advocating for not checking- as you say its not much of a warning for some additional checks and I think it might be helpful for those logging outputs to be able to see where the NaN hits. It could have some insight.